src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,213 @@
+package mate.jdbc.dao;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.jdbc.exception.DataProcessingException;
+import mate.jdbc.lib.Dao;
+import mate.jdbc.model.Car;
+import mate.jdbc.model.Driver;
+import mate.jdbc.model.Manufacturer;
+import mate.jdbc.util.ConnectionUtil;
+
+@Dao
+public class CarDaoImpl implements CarDao {
+    @Override
+    public Car create(Car car) {
+        String query = "INSERT INTO cars (model, manufacturer_id) VALUES (?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement saveCarStatement = connection.prepareStatement(query,
+                        Statement.RETURN_GENERATED_KEYS)) {
+            saveCarStatement.setString(1, car.getModel());
+            saveCarStatement.setLong(2, car.getManufacturer().getId());
+            saveCarStatement.executeUpdate();
+            ResultSet resultSet = saveCarStatement.getGeneratedKeys();
+            if (resultSet.next()) {
+                car.setId(resultSet.getObject(1, Long.class));
+            }
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Couldn't create car "
+                   + car + ". ", ex);
+        }
+        addAllDriversByCar(car);
+        return car;
+    }
+
+    @Override
+    public Optional<Car> get(Long id) {
+        String query = "SELECT c.id AS car_id, c.model, m.id AS manufacturer_id, m.name, m.county "
+                + "FROM cars c LEFT JOIN manufacturers m ON c.manufacturer_id = m.id "
```

Comment:
I think it`s better to use INNER JOIN instead LEFT JOIN here

Reply:
I need all data from the first table, so that's why LEFT JOIN